### PR TITLE
RateConverter: coalesce global reductions

### PR DESCRIPTION
**Pull request by akva2**
The main cost here is the latency of communication. Now we pay that price at most 2 times, instead of 9.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Review by atgeirr — COMMENTED** [review]
I think this is fine, although I needed to reread the logic a few times. Maybe add a comment or two, "// Calculating first using the hydrocarbon pore volumes.", "// No hydrocarbon pore volume, redo but this time using full pore volumes."

File: opm/simulators/wells/RateConverter.hpp [hunk 1]
```diff
@@ -614,24 +589,46 @@ namespace Opm {
...
-                double pressure;
-                double temperature;
-                double rs;
-                double rv;
-                double rsw;
-                double rvw;
-                double pv;
-                double saltConcentration;
+                Attributes(const Attributes& rhs)
+                    : data{0.0}
```
[the remark this answers is not in the document]
Comment (akva2) [reply opm/simulators/wells/RateConverter.hpp:604]: yes of course.  i always forget you can delegate as long as you don't extend the init list. i can add the comments, though i take no blame for them missing in the first place :p

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/wells/RateConverter.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (atgeirr) [reply opm/simulators/wells/RateConverter.hpp]: > i can add the comments, though i take no blame for them missing in the first place :p

The old logic was more clearly either-or than the new, that is why I requested them now.

**Pull request merged**